dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/resourcetable/table/DataSetOrganisationUnitCategoryResourceTable.java

```diff
@@ -97,29 +98,34 @@ public String getCreateTempTableStatement()
         {
             CategoryCombo categoryCombo = dataSet.getCategoryCombo();
 
+            Set<CategoryOption> availableOptions = getAvailableCategoryOptions( categoryCombo );
+
             for ( OrganisationUnit orgUnit : dataSet.getSources() )
             {
                 if ( !categoryCombo.isDefault() )
                 {
+                    Set<CategoryOption> orgUnitOptions = availableOptions;
```

Comment:
Note that this line will not create a new copy of `availableOptions`, it will just create a new pointer, hence adding items to this list will later iterations in the for loop. Is this intentional, or should we create a new copy of the list with `new HashSet<>()` ?

Reply:
yes, you are correct. we need to create a new copy of `availableOptions`